dm/tests/dmctl_command/run.sh

```diff
@@ -154,6 +155,71 @@ function run() {
 	fi
 }
 
+function checktask_full_mode_conn() {
+	# full mode
+	# dumpers: (2 + 2) for each
+	# loaders: 5 + 1 = 6
+	run_sql_source1 "set @@GLOBAL.max_connections=3;"
+	check_task_not_pass $cur/conf/dm-task3.yaml # dumper threads too few
+	run_sql_source1 "set @@GLOBAL.max_connections=4;"
+	check_task_pass $cur/conf/dm-task3.yaml
+
+	run_sql_tidb "set @@GLOBAL.max_connections=5;" # loader threads too few
+	check_task_not_pass $cur/conf/dm-task3.yaml
+	run_sql_tidb "set @@GLOBAL.max_connections=6;"
+	check_task_pass $cur/conf/dm-task3.yaml
+
+	run_sql_source1 "set @@GLOBAL.max_connections=151;"
+	run_sql_source2 "set @@GLOBAL.max_connections=151;"
+	run_sql_tidb "set @@GLOBAL.max_connections=151;"
+}
+
+function check_task_lightning() {
+	# unlimited connections: no need to warn
+	run_sql_tidb "set @@GLOBAL.max_connections=0;"
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"check-task $cur/conf/dm-task2.yaml" \
+		"\"passed\": true" 1 \
+		"task precheck cannot accurately check the number of connection needed for Lightning, please set a sufficiently large connections for TiDB" 0
+	run_sql_tidb "set @@GLOBAL.max_connections=5;"
+	# fail but give warning, because it's using Lightining
+	run_dm_ctl $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+		"check-task $cur/conf/dm-task2.yaml" \
+		"task precheck cannot accurately check the number of connection needed for Lightning, please set a sufficiently large connections for TiDB" 1
+}
+
+function check_full_mode_conn() {
+	# TODO: currently, pool-size are not efficacious for Lightning
+	# which simply determines the concurrency by hardware conditions.
+	# This should be solved in the future.
+	run_sql_tidb "set @@GLOBAL.max_connections=151;"
```

Comment:
we can limit the connection to 4 and 6 respectively, to test `getConfigConn` is correct

Reply:
All these tests are latently checking the correctness of `getConfigConn`. For example, in `checktask_full_mode_conn`, when I set the `max_connections` to 3, the checker can fail, whilst, when I set it to 4, the checker can pass. In this sense, the checker-or more specifically, the `getConfigConn`-is expected to return the value 4 with respect to our config setting, and whether it passes this test or not categorically indicates if it can work accurately.

And in terms of this test, `check_full_mode_conn`, has a disparate intention to evaluate whether the number of connections established between DM and the databases is precisely what we expect beforehand.

Reply:
OK I think SHOW PROCESSLIST and check_rows_equal should be enough